File: Find Closest Number to Zero - Leetcode 2239/Find Closest Number to Zero - Leetcode 2239.java
import java.util.HashSet;
import java.util.Set;

public class Solution {
    public int findClosestNumber(int[] nums) {
        int closest = nums[0];
        
        for (int x : nums) {
            if (Math.abs(x) < Math.abs(closest)) {
                closest = x;
            }
        }
        
        if (closest < 0 && contains(nums, Math.abs(closest))) {
            return Math.abs(closest);
        } else {
            return closest;
        }
    }

    private boolean contains(int[] nums, int value) {
        for (int num : nums) {
            if (num == value) {
                return true;
            }
        }
        return false;
    }
}

File: Find Closest Number to Zero - Leetcode 2239/Find Closest Number to Zero - Leetcode 2239.py
from typing import List

class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        closest = nums[0]
        for x in nums:
            if abs(x) < abs(closest):
                closest = x
        
        if closest < 0 and abs(closest) in nums:
            return abs(closest)
        else:
            return closest
        
        # Time: O(n)
        # Space: O(1)
